src/Orc.DataAccess.Xaml/Controls/ConnectionStringBuilder/Views/ConnectionStringBuilder.cs

```diff
@@ -50,10 +51,10 @@ private async void OnEditCommandExecuted(object sender, ExecutedRoutedEventArgs
             using (new DisposableToken<ConnectionStringBuilder>(this, x => x.Instance.SetCurrentValue(IsInEditModeProperty, true), 
                 x => x.Instance.SetCurrentValue(IsInEditModeProperty, false)))
             {
-                var connectionStringEditViewModel = _typeFactory.CreateInstanceWithParametersAndAutoCompletion<ConnectionStringEditViewModel>(ConnectionString, _dbProvider);
+                var connectionStringEditViewModel = _typeFactory.CreateRequiredInstanceWithParametersAndAutoCompletion<ConnectionStringEditViewModel>(ConnectionString, _dbProvider);
                 connectionStringEditViewModel.IsAdvancedOptionsReadOnly = IsAdvancedOptionsReadOnly;
 
-                if (await _uiVisualizerService.ShowDialogAsync(connectionStringEditViewModel) ?? false)
+                if ((await _uiVisualizerService.ShowDialogAsync(connectionStringEditViewModel)).DialogResult ?? false)
```

Comment:
as coding standard, I think we should store the result in a variable. Code is not incorrect, but easier to maintain / read in the future if we store in variable.